doc/configuration.md

```diff
@@ -4,48 +4,329 @@ This document describes how to configure tapyrus-signer.
+* `aggregated-public-key`
+This is required.
+This is the public key, which can be used to verify block proofs.
+This public key is aggregate of all federation member's public keys.  
+* `node-vss`
+This is optional.
+Verifiable Secret Share and commitments from all signers in the federation.
+This field may be empty when the signer is not a member of the federation.
+This item should not specify if the signer is not a member of the federation.
+See also [Tapyrus signer network paramters](doc/setup.md#tapyrus-signer-network-parameters).
+
+Here describe some `federations.toml` examples for particular scenarios.
+
+### Scenario 1: The signer is a member of the first federation of the chain.
 
-- `redis-host`: Redis Server host name or IP Address.
+The signers who are a member of the first federation of the chain must set the first federation parameters into their `federations.toml`.
+If the federation has five members, and the threshold is 3, then the `federations.toml` would be like below.
+
+```toml
+[[federation]]
+block-height = 0
+threshold = 3
+aggregated-public-key = "030d856ac..."
+node-vss = [
+  "02472012cf49fca573ca...",
+  "02785a891f323acd6cef...",
+  "02ce7edc292d7b747fab...",
+  "02d111519ba1f3013a7a...",
+  "03831a69b8009833ab5b..."
+]
+```
+
+If you don't have Node VSSs and Aggregate public key, you should follow steps in [Generate Aggregate public key and Node secret share for Tapyrus-signer network](doc/setup.md#generate-aggregate-public-key-and-node-secret-share-for-tapyrus-signer-network)
+
+## Scenario 2: The signer is out of the federation member from future block height.
+
+Here assume the current block height of the chain tip is somewhere among 0 to 98.
+The signer is a member of the current federation.
+The federation is scheduled to transition to the new federation from 100 height.
+
+```toml
+[[federation]]
+block-height = 0
+threshold = 3
+aggregated-public-key = "030d856ac..."
+node-vss = [
+  "02472012cf49fca573ca...",
+  "02785a891f323acd6cef...",
+  "02ce7edc292d7b747fab...",
+  "02d111519ba1f3013a7a...",
+  "03831a69b8009833ab5b..."
+]
+
+[[federation]]
+block-height = 100
+aggregated-public-key = "030acd6af9..."
+```
+
+After generating 99 height block proof, the signer can stop its own tapyrus-signerd.
+
+## Scenario 3: The signer joins the federation from future block height.
+
+Here assume the current block height of the chain tip is somewhere among 0 to 98.
+The signer is not a member of the current federation.
+The federation is scheduled to transition to the new federation from 100 height.
+
+```toml
+[[federation]]
+block-height = 0
+aggregated-public-key = "030d856ac..."
+
+[[federation]]
+block-height = 100
+threshold = 2
+aggregated-public-key = "030acd6af9..."
+node-vss = [
+  "02472012cf49fca573ca...",
+  "02ce7edc292d7b747fab...",
+  "03831a69b8009833ab5b..."
+]
+```
+
+Before generating 99 height block proof, you should startup your tapyrus-signerd daemon. 
```

Comment:
I think `Before generating 100 height block proof` is correct.
This member can't participate 99 generation and so any messages is not sent from other signer at 99.

Reply:
It's true. I will fix. Thanks.

Reply:
fixed at 437acaf62e2bce035dfb35cfd4f53da556b5d9af